File: ios/PiTools/Sources/PiTools/BuiltInTools.swift
import Foundation
import PiAgentCore

public struct BuiltInTools {
    /// Create all built-in tools with the given sandbox URL.
    public static func create(sandboxURL: URL) -> [any Tool] {
        [
            ReadFileTool(sandboxURL: sandboxURL),
            WriteFileTool(sandboxURL: sandboxURL),
            EditFileTool(sandboxURL: sandboxURL),
            ListFilesTool(sandboxURL: sandboxURL),
            SqliteQueryTool(),
            HttpRequestTool(),
            MediaQueryTool(),
            ExternalFileTool(),
        ]
    }
}

File: android/pi-tools/src/main/kotlin/com/pimobile/tools/BuiltInTools.kt
package com.pimobile.tools

import android.content.Context
import com.pimobile.agent.Tool
import java.io.File

object BuiltInTools {
    fun create(context: Context, sandboxDir: File): List<Tool> = listOf(
        ReadFileTool(context.contentResolver, sandboxDir),
        WriteFileTool(sandboxDir),
        EditFileTool(sandboxDir),
        ListFilesTool(sandboxDir),
        SqliteQueryTool(),
        HttpRequestTool(),
        MediaQueryTool(context.contentResolver),
        ExternalFileTool()
    )
}
